mypy/plugins/enums.py

```diff
@@ -165,19 +167,26 @@ class SomeEnum:
                 get_proper_type(n.type) if n else None
                 for n in stnodes
                 if n is None or not n.implicit)
-            proper_types = (
+            proper_types = list(
                 _infer_value_type_with_auto_fallback(ctx, t)
                 for t in node_types
                 if t is None or not isinstance(t, CallableType))
             underlying_type = _first(proper_types)
             if underlying_type is None:
                 return ctx.default_attr_type
+
             all_same_value_type = all(
                 proper_type is not None and proper_type == underlying_type
                 for proper_type in proper_types)
             if all_same_value_type:
                 if underlying_type is not None:
                     return underlying_type
+
+            all_equivalent_types = all(
+                proper_type is not None and is_equivalent(proper_type, underlying_type)
+                for proper_type in proper_types)
+            if all_equivalent_types:
+                return make_simplified_union(cast(Sequence[Type], proper_types))
```

Comment:
It's unclear to me when the above new code is needed. Is there a test case that depends on this? If not, could you add a test case. 

In any case, it would be good to have a comment here.

Reply:
Here are related changes:
1. https://github.com/python/mypy/pull/10852/files#diff-ebc735ad8fa6eb42d5390fa4fa9ef60581764c7bbfb5eafcde16f5e40c330c84R69
2. https://github.com/python/mypy/pull/10852/files#diff-ebc735ad8fa6eb42d5390fa4fa9ef60581764c7bbfb5eafcde16f5e40c330c84R143
3. https://github.com/python/mypy/pull/10852/files#diff-ebc735ad8fa6eb42d5390fa4fa9ef60581764c7bbfb5eafcde16f5e40c330c84R102

Reply:
Updated code with more comments 👍